server/src/main/java/org/opensearch/index/query/RangeQueryBuilder.java

```diff
@@ -58,6 +59,7 @@
  *
  * @opensearch.internal
  */
+@PublicApi(since = "1.0.0")
```

Comment:
IIUC this is meant to be on the interface or some very generic classes. 
Eitherways since other query builders are also not annotated with this should we not include as part of this PR ?

Reply:
Ideally yes. You are correct.

This is a required workaround since I needed `RangeQueryBuilder` as an argument in an experimental API.
I can add a TODO & open up an issue as well to remove this annotation once we take out star-tree from experimental (which is also planned in 3.1).

What do you say?

Reply:
In case there are delays in https://github.com/opensearch-project/OpenSearch/pull/18070, will raise a follow-up PR to create a new POJO to replace usage of RangeQueryBuilder.